.circleci/config.yml

```diff
@@ -37,10 +72,147 @@ jobs:
           name: 'Ensure clean dist diff'
           command: make validate-no-changes
 
+  build-docs:
+    executor: default_build_environment
+    working_directory: ~/repo
+    
+    steps:
+      - checkout
+      - base-tools/install
+      - bootstrap_node
+      - bootstrap_pipenv
+      - run:
+          name: 'Typedoc'
+          command: make docs
+      - run:
+          name: "MkDocs"
+          command: make build-mkdocs
+      - persist_to_workspace:
+          root: ~/repo
+          paths:
+            - site
+
+  push-adhoc-docs:
+    executor: default_build_environment
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+      - bootstrap_node
+      - bootstrap_aws_cli
+      - attach_workspace:
+          at: ~/repo
+      - run:
+          name: Deploy Adhoc
+          command: 'make publish-docs-adhoc FLAGS=--forceUpload'
```

Comment:
we force upload since we are only doing this on release, so we should be careful about merging into release instead of breaking the pipeline whenever we need to make changes to an existing version

Reply:
can you say a bit more about "we should be careful about merging into release instead of breaking the pipeline whenever we need to make changes to an existing version"?